rust/ip-packet/src/buffer_pool.rs

```diff
@@ -0,0 +1,96 @@
+use std::{
+    ops::{Deref, DerefMut},
+    sync::{Arc, LazyLock},
+};
+
+use crate::MAX_DATAGRAM_PAYLOAD;
+
+type BufferPool = Arc<lockfree_object_pool::MutexObjectPool<Vec<u8>>>;
+
+static BUFFER_POOL: LazyLock<BufferPool> = LazyLock::new(|| {
+    Arc::new(lockfree_object_pool::MutexObjectPool::new(
+        || vec![0; MAX_DATAGRAM_PAYLOAD],
+        |v| v.fill(0),
+    ))
+});
```

Comment:
Buffer pool defined as `static` memory so it will live for however long our process lives / we are loaded into RAM.